pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -376,6 +470,232 @@ def _reraise_if_not_permitted(self, err: VerificationError, code: str, permits:
         self._logger.warn("Issue %s is permitted with justification: %s", err, justification)
         return justification
 
+    async def publish_client(self, working_dir, from_release_tag, release_name, arch, client_type):
+        subprocess.run(f"docker login -u openshift-release-dev+art_quay_dev -p {os.environ['PASSWORD']} quay.io")
+        minor = release_name.split(".")[1]
+        quay_url = constants.QUAY_URL
+        # Anything under this directory will be sync'd to the mirror
+        BASE_TO_MIRROR_DIR = f"{working_dir}/to_mirror/openshift-v4"
+        shutil.rmtree(BASE_TO_MIRROR_DIR, ignore_errors=True)
+
+        # From the newly built release, extract the client tools into the workspace following the directory structure
+        # we expect to publish to mirror
+        CLIENT_MIRROR_DIR = f"{BASE_TO_MIRROR_DIR}/{arch}/clients/{client_type}/{release_name}"
+        os.makedirs(CLIENT_MIRROR_DIR)
+
+        if arch == 'x86_64':
+            # oc image  extract requires an empty destination directory. So do this before extracting tools.
+            # oc adm release extract --tools does not require an empty directory.
+            image_stat, oc_mirror_pullspec = get_release_image_pullspec(f"{quay_url}:{from_release_tag}", "oc-mirror")
+            if image_stat == 0:  # image exist
+                # extract image to workdir, if failed it will raise error in function
+                extract_release_binary(oc_mirror_pullspec, f"--path=/usr/bin/oc-mirror:{CLIENT_MIRROR_DIR}")
+                # archive file
+                with tarfile.open(f"{CLIENT_MIRROR_DIR}/oc-mirror.tar.gz", "w:gz") as tar:
+                    tar.add(f"{CLIENT_MIRROR_DIR}/oc-mirror")
+                # calc shasum
+                with open(f"{CLIENT_MIRROR_DIR}/oc-mirror.tar.gz", 'rb') as f:
+                    shasum = hashlib.sha256(f.read()).hexdigest()
+                # write shasum to sha256sum.txt
+                with open(f"{CLIENT_MIRROR_DIR}/sha256sum.txt", 'a') as f:
+                    f.write(f"{shasum} oc-mirror.tar.gz")
+                # remove oc-mirror
+                os.remove(f"{CLIENT_MIRROR_DIR}/oc-mirror")
+
+        # extract release clients tools
+        extract_release_client_tools(f"{quay_url}:{from_release_tag}", f"--to={CLIENT_MIRROR_DIR}", None)
+        # create symlink for clients
+        self.create_symlink(CLIENT_MIRROR_DIR, False, False)
+
+        if minor > 0:
```

Comment:
minor is a string above, comparison will fail. more to the point, when will minor ever not be >0?

Reply:
the logic comes from https://github.com/openshift-eng/aos-cd-jobs/blob/master/pipeline-scripts/release.groovy#L560
I guess it was used for making sure `minor - 1` works in the following code, I think this check is useless now, maybe it was introduced in ancient 4.0

Reply:
Let's remove it